command/tls_cert_create.go

```diff
@@ -0,0 +1,289 @@
+package command
+
+import (
+	"crypto/x509"
+	"fmt"
+	"io/ioutil"
+	"net"
+	"strings"
+
+	"github.com/hashicorp/nomad/helper/flags"
+	"github.com/hashicorp/nomad/helper/tlsutil"
+	"github.com/hashicorp/nomad/lib/file"
+	"github.com/posener/complete"
+)
+
+type TLSCertCreateCommand struct {
+	Meta
+}
+
+func NewCertCreate() *TLSCertCreateCommand {
+	return &TLSCertCreateCommand{}
+}
+
+func (c *TLSCertCreateCommand) Help() string {
+	helpText := `
+Usage: nomad tls cert create [options]
+
+Create a new certificate
+
+$ nomad tls cert create -server
+==> WARNING: Server Certificates grants authority to become a
+    server and access all state in the cluster including root keys
+    and all ACL tokens. Do not distribute them to production hosts
+    that are not server nodes. Store them as securely as CA keys.
+==> Using nomad-agent-ca.pem and nomad-agent-ca-key.pem
+==> Saved dc1-server-nomad-0.pem
+==> Saved dc1-server-nomad-0-key.pem
+$ nomad tls cert create -client
+==> Using nomad-agent-ca.pem and nomad-agent-ca-key.pem
+==> Saved dc1-client-nomad-0.pem
+==> Saved dc1-client-nomad-0-key.pem
+
+Certificate Create Options:
+  -additional-dnsname
+    Provide an additional dnsname for Subject Alternative Names.
+    localhost is always included. This flag may be provided multiple times.
+
+  -additional-ipaddress
+    Provide an additional ipaddress for Subject Alternative Names.
+    "127.0.0.1 is always included. This flag may be provided multiple times.")
+
+  -ca
+    Provide path to the ca. Defaults to #DOMAIN#-agent-ca.pem.
+
+  -cli
+    Generate cli certificate.
+
+  -client
+    Generate client certificate.
+
+  -days
+    Provide number of days the certificate is valid for from now on. 
+    Defaults to 1 year.
+
+  -dc
+    Provide the datacenter. Matters only for -server certificates. 
+    Defaults to dc1.
+
+  -domain
+    Provide the domain. Matters only for -server certificates.
+  
+  -key
+    Provide path to the key. Defaults to #DOMAIN#-agent-ca-key.pem.
+
+  -node
+    When generating a server cert and this is set an additional dns name is 
+    included of the form <node>.server.<datacenter>.<domain>.
```

Comment:
I find this flag name confusing as we also use the term interchangeably to describe clients.

Reply:
Do we need this granularity? I can remove it

Reply:
After thinking about this I have removed the option for this.